### PR TITLE
HAAR-865 user search by email

### PR DESCRIPTION
**Pull request by AlTurner-MOJ**

### CODE REVIEW

File: src/test/kotlin/uk/gov/justice/digital/hmpps/manageusersapi/resource/ExternalUserSearchControllerIntTest.kt [hunk 1]
```diff
@@ -0,0 +1,62 @@
...
+
+import org.assertj.core.api.Assertions.assertThat
+import org.junit.jupiter.api.Nested
+import org.junit.jupiter.api.Test
+import uk.gov.justice.digital.hmpps.manageusersapi.integration.IntegrationTestBase
+
+class ExternalUserSearchControllerIntTest : IntegrationTestBase() {
+
+  @Nested
+  inner class FindUsersByEmail {
```
Comment (ravmoj) [inline src/test/kotlin/uk/gov/justice/digital/hmpps/manageusersapi/resource/ExternalUserSearchControllerIntTest.kt:11]: is there any 'isNotFound' scenarios here?

Comment (AlTurner-MOJ) [reply src/test/kotlin/uk/gov/justice/digital/hmpps/manageusersapi/resource/ExternalUserSearchControllerIntTest.kt:11]: The end point in external users mimics the existing behaviour of the corresponding Auth end point, which gives a no content response when no match is found, and also when an email is not supplied

File: src/main/kotlin/uk/gov/justice/digital/hmpps/manageusersapi/resource/ExternalUserSearchController.kt [hunk 2]
```diff
@@ -0,0 +1,114 @@
...
+        responseCode = "401",
+        description = "Unauthorized.",
+        content = [
+          Content(
+            mediaType = "application/json",
+            schema = Schema(implementation = ErrorResponse::class)
+          )
+        ]
+      ),
+      ApiResponse(
```
Comment (ravmoj) [inline src/main/kotlin/uk/gov/justice/digital/hmpps/manageusersapi/resource/ExternalUserSearchController.kt:51]: Don't need 404  here , instead pls add 204

**Review by ravmoj — APPROVED** [review]
lgtm

**Pull request merged**